src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -89,78 +89,43 @@ def on_simulation_end(self, event: Event) -> None:
-        self.population_view = builder.population.get_view(columns=[])
-        self.response_probability_decennial = builder.lookup.build_table(
-            data=data_values.RESPONSE_PROBABILITY_DECENNIAL
-        )
+        
+    def get_population_view(self, builder) -> PopulationView:
+        """Get the population view to be used for observations"""
+        return builder.population.get_view(columns=metadata.DECENNIAL_CENSUS_COLUMNS_USED)
 
-        # FIXME: register to happen "on_collect_metrics" (end of time step)
-        builder.event.register_listener("time_step__prepare", self.on_time_step__prepare)
-        builder.event.register_listener("simulation_end", self.on_simulation_end)
-
-    def on_time_step__prepare(self, event: Event) -> None:
-        if self.counter == 0:
-            start_date_str = (
-                f"ymd_{self.start_date.year}_{self.start_date.month}_{self.start_date.day}"
-            )
-            state_table = self.population_view.get(event.index)
-            state_table.to_hdf(self.output_path, start_date_str)
-        self.counter += 1
-        if (self.clock().year % 10 == 0) & (self.clock().month == 4):
-            if self.clock().day < 29:  # because we only want one observation in April
-                self.decennial_census(event, hdf_key=f"year_{self.clock().year}")
+    def get_responses(self) -> pd.DataFrame:
+        return pd.DataFrame()  # TODO: include schema here, including column for census year
 
-    def on_simulation_end(self, event: Event) -> None:
-        end_date_str = f"ymd_{self.end_date.year}_{self.end_date.month}_{self.end_date.day}"
-        state_table = self.population_view.get(event.index)
-        state_table.to_hdf(self.output_path, end_date_str)
+    def to_observe(self, event: Event) -> bool:
+        if (self.clock().year % 10 == 0) & (self.clock().month == 4):
+            if self.clock().day < 29:  # because we only want one observation in April  FIXME: cooler to do this with the timestep
+                return True
 
-    def decennial_census(self, event: Event, hdf_key) -> None:
-        pop = self.population_view.subview(metadata.DECENNIAL_CENSUS_COLUMNS_USED).get(
-            event.index
+    def do_observation(self, event) -> None:
+        pop = self.population_view.get(
+            event.index, # query="alive == 'alive'",  # TODO: uncomment this to include only living simulants in census
         )
         pop["middle_initial"] = pop["middle_name"].astype(str).str[0]
         pop = pop.drop(columns="middle_name")
 
-        # we don't have a 100% census response rate:
-        respondents = self.randomness.filter_for_probability(
-            pop, self.response_probability_decennial(pop.index), "decennial_respondents"
-        )
+        # TODO: include additional columns specified in MIC-3642
 
-        respondents.to_hdf(self.decennial_path, hdf_key)
+        self.responses = pd.concat([self.responses, pop])
```

Comment:
As is, this is going to save out a single hdf with a single default key. The only way to differentiate which census is by looking at the date column. Maybe this is fine?

Another option is to get clever with `on_simulation_end` and wrap the saving to hdf part in a method that can be overwritten. Then in this class you can subset to however many cencus' you have in the object and save out w/ different keys. Thoughts?

@rmudambi

Reply:
I'm going to save all censuses in one table for now, but Rajan suggested that I could override on_simulation_end if we decided that we wanted them to be in separate files.